web/src/routes/~_dashboard/ReportIssueButton.tsx

```diff
@@ -0,0 +1,207 @@
+import { SideLabel } from "@stanfordspezi/spezi-web-design-system/components/SideLabel";
+import { cn } from "@stanfordspezi/spezi-web-design-system/utils/className";
+import { useOpenState } from "@stanfordspezi/spezi-web-design-system/utils/useOpenState";
+import { Flag } from "lucide-react";
+import { z } from "zod";
+import { useState } from "react";
+import { addDoc, collection } from "firebase/firestore";
+import { firestore, getCurrentUser } from "@/modules/firebase/app";
+import { toast } from "@stanfordspezi/spezi-web-design-system/components/Toaster";
+
+const issueStrings = [
+  "The content is dangerous or harmful",
+  "The content is inaccurate or misleading",
+  "Important details or key words are missing",
+  "The explanations or follow-up questions are unclear",
+  "I'm experiencing technical issues",
+];
+const justPredefinedIssuesSchema = z.object({
+  issueSelection: z
+    .array(z.boolean())
+    .length(issueStrings.length)
+    .refine((userSelection) => userSelection.some((value) => value === true), {
+      message: "At least one issue has to be selected",
+    }),
+  isOtherSelected: z.literal(false),
+  userInputedIssue: z.string(),
+});
+const otherSelectedSchema = z.object({
+  issueSelection: z.array(z.boolean()).length(issueStrings.length),
+  isOtherSelected: z.literal(true),
+  userInputedIssue: z
+    .string()
+    .min(1, { message: "Please specify the issue in more detail" }),
+});
+
+const formSchema = z.union([justPredefinedIssuesSchema, otherSelectedSchema]);
+
+interface UserIssueContext {
+  reportID: string;
+  observationIndex?: number;
+  explanation?: boolean;
+  questionIndex?: number;
+}
+
+interface ReportIssueButtonProps {
+  className?: string;
+  context: UserIssueContext;
+}
+
+export function ReportIssueButton({
+  className,
+  context,
+}: ReportIssueButtonProps) {
+  const openState = useOpenState(false);
+  const defaultValues = {
+    issueSelection: new Array(issueStrings.length).fill(false),
+    isOtherSelected: false,
+    userInputedIssue: "",
+  };
+
+  const form = useForm({
+    formSchema,
+    defaultValues: defaultValues,
+  });
+  const [otherExpanded, setOtherExpanded] = useState(false);
+
+  const handleSubmit = form.handleSubmit(async (medicalReport) => {
+    const selectedIssueStrings = medicalReport.issueSelection
+      .map((isUserSelected, index) =>
+        isUserSelected ? issueStrings[index] : null,
+      )
+      .filter((val) => val !== null);
+
+    if (
+      selectedIssueStrings ||
+      (medicalReport.isOtherSelected && medicalReport.userInputedIssue)
+    ) {
+      let issueContent = {};
+      if (selectedIssueStrings) {
+        issueContent = {
+          ...issueContent,
+          pre_defined_issues: selectedIssueStrings,
+        };
+      }
+      if (medicalReport.isOtherSelected) {
+        issueContent = {
+          ...issueContent,
+          user_inputed_issue: medicalReport.userInputedIssue,
+        };
+      }
```

Comment:
Equivalent, but removed mutation. More deterministic and easier to follow.

```suggestion
      const issueContent = selectedIssueStrings? {
          pre_defined_issues: selectedIssueStrings,
        } : medicalReport.isOtherSelected ? {
          user_inputed_issue: medicalReport.userInputedIssue,
        } : {}
```

Reply:
I addressed it